boot/nxboot/loader/boot.c

```diff
@@ -403,6 +440,42 @@ static int perform_update(struct nxboot_state *state, bool check_only)
   return OK;
 }
 
+#ifdef CONFIG_NXBOOT_COPY_TO_RAM
+int nxboot_ramcopy(void)
+{
+  int primary;
+  struct nxboot_img_header header;
+  ssize_t bytes;
+  static uint8_t *buf;
+  size_t size;
+
+  primary = flash_partition_open(CONFIG_NXBOOT_PRIMARY_SLOT_PATH);
+  if (primary < 0)
+    {
+      return ERROR;
+    }
+
+  get_image_header(primary, &header);
+  buf = malloc(header.size);
+  if (!buf)
+    {
+      return ERROR;
+    }
+
+  size = header.size - CONFIG_NXBOOT_HEADER_SIZE;
+  bytes = pread(primary, buf, size, CONFIG_NXBOOT_HEADER_SIZE);
+  if (bytes != size)
+    {
+      return ERROR;
+    }
+
+  memcpy((uint32_t *)CONFIG_NXBOOT_RAMSTART, buf, size);
+  flash_partition_close(primary);
+
```

Comment:
`free(buf)`

Reply:
@michallenc Would be the thorough approach for sure. In will be "zapped" when the loaded image is run but, to be fair, that is probably only in my case where it loads a complete binary included full initialization etc. I will add it.

Reply:
done